### ISSUE TITLE
How to only download the file if the filename doesn't already exist

### ISSUE THREAD

**Issue by CMCDragonkai** [aria2/aria2#865]
I passed in an input list of urls, these urls have specific names. Sometimes I'll update the url list with new urls. I don't want aria2c to download urls which have already been downloaded.

I looked at the `--conditional-get` flag, however this flag only works if the server supports If-Modified-By, and also I don't care about the date of a package, only whether the name of the file already exists in the download directory.

What's the best way to do this?

**Comment by kcolford** [comment]
Use the `-c` option to _continue existing downloads_ which will not redownload anything. I have this set to an alias since it's so useful.

**Comment by CMCDragonkai** [comment]
@kcolford 

The docs says that option does:

> Continue downloading a partially downloaded file. Use this option to resume a download started by a web browser or another program which downloads files sequentially from the beginning. Currently this option is only applicable to HTTP(S)/FTP downloads.

So it also doesn't download if the file is already "downloaded"? That should be added to the docs!

**Comment by kcolford** [comment]
Not all sequential downloaders leave a .part file lying around. So the most general thing is to check if the whole file is there and if not, just download chunks to finish the file. It's not perfect though, downloading github releases doesn't work quite right because the "Content-Disposition" header sets a particular filename that confuses aria2, but I'm pretty sure there's an issue for that in the works right now...

Maybe it should be documented better, I suggest submitting a patch for it since it's just documentation and doesn't require much technical skill.

**Comment by SumatraPeter** [comment]
Simply adding **--auto-file-renaming=false** should be sufficient IMO. If a control file (.aria2) exists then the download will resume, otherwise if no control file exists then since **--allow-overwrite** is **false** by default the file will not be re-downloaded.

**Comment by kcolford** [comment]
@sumatrapeter ah, that's a good suggestion, I think I'll add it to my alias, hopefully we can leave this issue open as documentation

**Comment by fcgouveia** [comment]
Thanks a lot for that!

**Comment by monktastic** [comment]
Yes, thanks! "-c" didn't do what I wanted but "--auto-file-renaming=false" seems to.

**Comment by NightMachinary** [comment]
@kcolford Why is an open issue a better fit for documentation?

**Comment by kcolford** [comment]
@nightmachinary because it's simple and convenient, see K.I.S.S.

If someone wants to go ahead and improve the docs or an FAQ, then go ahead so we can close this issue. But aria2 is still being largely developed so I don't think anyone is interested at this point.

**Comment by lordkian** [comment]
Hi to everyone.
the combination of `--auto-file-renaming=false` `--allow-overwrite=false` worked for me.
I know its dummy but can you mention to use "=" in the man page? it didn't work for me at first because I used `--allow-overwrite false`.
